icu4c/source/tools/toolutil/writesrc.cpp

```diff
@@ -357,6 +358,69 @@ usrc_writeUCPTrie(FILE *f, const char *name, const UCPTrie *pTrie, UTargetSyntax
     usrc_writeUCPTrieStruct(f, line, pTrie, line2, line3, line4, syntax);
 }
 
+U_CAPI void U_EXPORT2
+usrc_writeUnicodeSet(
+        FILE *f,
+        const char *name,
+        const USet *pSet,
+        UTargetSyntax syntax) {
+    // ccode is not yet supported
+    U_ASSERT(syntax == UPRV_TARGET_SYNTAX_TOML);
+
+    fprintf(f, "name = \"%s\"\n", name);
+
+    // First write as the uset serialized format
```

Comment:
Is that what we want for ICU4X?
This format is used in some ICU4C data files, designed for serialization into an array of uint16_t.
As such, it's more compact for BMP code points (1 uint16_t each), and it's usable with a specialized contains(code point) without deserializing/heap allocation, but it's different from the ICU4X UnicodeSet inversion list format.

Note that you asked in today's meeting with Iain what the unit type should be for the "serialized" array. I said uint32_t assuming you just wrote range limits, but for this specific output it's actually uint16_t, as you can see from your code here.

If ICU4X does not need this, then I would just drop it.

Reply:
13ee49a37aa6e612a3ac5258f4b925695d336d57